transcendence-app/migrations/20221206105158_add_moderators_table.ts

```diff
@@ -0,0 +1,18 @@
+import { Knex } from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+  return knex.raw(`
+    CREATE TABLE
+      IF NOT EXISTS Moderators (
```

Comment:
thought: I'm concerned this table schema will be hard to expand on later. For example, if we want to add a `banned` role.

question: What do you think about creating a `Roles` table with the role name as its `id` where we could insert the roles and an `UsersToRoles` where we assign roles to users?

```mermaid
erDiagram
    Users |o--o{ UsersToRoles: has
    Users {
        uuid id PK "The user ID"
    }
    UsersToRoles {
        uuid userId FK "The user ID"
        text roleId FK "The role ID"
    }
    UsersToRoles }o--o| Roles: has
    Roles {
        text id PK "The role ID (name of the role)"
    }

```

Reply:
I'm not sure I get it. Let's discuss it on Saturday when we meet.

Reply:
Are we sure the roles couldn't be just an enum type, instead of a full blown table?

Reply:
An enum for the Roles could also work :+1:

Reply:
I think it's better as you proposed. Like this, we can do an on delete cascade in the tables that reference those roles. I didn't know how to do the same with enums, although maybe there is a way. I implemented it as you say. Please, take a look again.